tests/core/test_core.py

```diff
@@ -116,35 +116,65 @@ def test_narrow_to_stream(self, mocker, controller,
         id_list = index_stream['stream_msg_ids_by_stream_id'][stream_id]
         assert {widget.original_widget.message['id']} == id_list
 
-    def test_narrow_to_topic(self, mocker, controller,
-                             msg_box, index_topic):
-        expected_narrow = [['stream', msg_box.stream_name],
-                           ['topic', msg_box.topic_name]]
-        controller.model.narrow = []
+    @pytest.mark.parametrize('initial_narrow, final_narrow, anchor', [
+        ([], [['stream', 'BOO'], ['topic', 'RDS']], None),
```

Comment:
To clarify, my intention behind keeping the final narrow same was that, the `index_topic` fixture from `conftest` contains just one topic and one message, so having multiple final narrows would need some serious refactoring.

No, we don't gain anything from exploring initial_narrow state space, it was just to assure that narrowing works for all possible combinations of initial narrow.

I'll have to explore more on how I can improve the tests with minimal change. Also, breaking the tests seems very unlikely since the index contains just one message and it would focus on that even if we provide an anchor or don't.